src/app/shared/services/auth.service.ts

```diff
@@ -1,44 +1,54 @@
 import { Injectable } from '@angular/core';
 import { User } from '@shared/types/user';
-import { BehaviorSubject, map, Observable, of } from 'rxjs';
-import { UserMock } from '@app/testing/user.mock';
+import { BehaviorSubject, catchError, filter, map, Observable, of, switchMap, tap } from 'rxjs';
 import { Router } from '@angular/router';
+import { HttpClient } from '@angular/common/http';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthService {
 
-  private _currentUser: BehaviorSubject<User> = new BehaviorSubject<User>(null);
+  private _isAuthenticated = new BehaviorSubject<boolean>(false);
 
   constructor(
-    private router: Router
+    private router: Router,
+    private http: HttpClient
   ) { }
 
-  login(email: string, password: string): Observable<boolean> {
-    const user = UserMock;
-    if (email !== user.email || password !== user.password)
-      return of(false);
-    localStorage.setItem('currentUser', JSON.stringify(user));
-    this._currentUser.next(user);
-    return of(true);
+  login(login: string, password: string): Observable<{token: string | null}> {
+    return this.http.post<{token: string}>('auth/login', {
+      login, password
+    }).pipe(
+      tap(res => {
+        localStorage.setItem('userToken', res.token);
+        this._isAuthenticated.next(true);
+      }),
+      catchError(() => of({token: null}))
+    );
   }
 
   logout(): void {
-    localStorage.removeItem('currentUser');
-    this._currentUser.next(null);
+    localStorage.removeItem('userToken');
+    this._isAuthenticated.next(false);
     this.router.navigate(['/auth']);
   }
 
   isAuthenticated(): Observable<boolean> {
-    const currentUser = localStorage.getItem('currentUser');
-    if (currentUser) this._currentUser.next(JSON.parse(currentUser));
-    return this._currentUser.asObservable().pipe(
-      map(res => !!res)
+    return this._isAuthenticated.asObservable().pipe(
+      map(() => !!localStorage.getItem('userToken'))
```

Comment:
you are not using _isAuthenticated vakue here at all. So maybe you could remove it at all? for example make return of(!!localStorage.getItem('userToken')); 
Or maybe you can syncronize _isAuthenticated state with localStorage in constructor (when app starts)

Reply:
fixed